### ISSUE TITLE
Options for filename patterns

### ISSUE THREAD

**Issue by tapioca24** [tapioca24/p5.capture#12]
## v1.2.0 specification

| file format                 | filename pattern              | example                |
| --------------------------- | ----------------------------- | ---------------------- |
| video (webm, gif, mp4), zip | timestamp (`YYYYMMDD-HHmmdd`) | `20220729-003612.webm` |
| image (png, jpg, webp)      | 7-digit numbers               | `0000123.png`          |

The timestamp is to avoid duplicate file names.

## Issues

The following issues are considered.

- video, zip
  - Arbitrary strings cannot be included in the file name
  - Some locales are not comfortable with the datetime format `YYYYYMMDD-HHmmdd`
- image
  - Arbitrary strings cannot be included in the file name
  - Overflow of digits in long animations
  - May not want the counter to reset on each run

## Requirements

It is cool if the following requirements are met.

- Avoid duplicate file names
- Can include any string in the file name
- Can specify the number of digits in the counter
- Can specify a starting value for the counter

**Comment by tapioca24** [comment]
There are two options.

- Objects to customize filename parameters
- Functions that return a filename

## Objects to customize filename parameters

For example, something like this:

```js
P5Capture.setDefaultOptions({
  baseFilename: {
    pattern: "my-sketch-<timestamp>",
    timestampFormat: "MMM-DD-YYYY-HHmmss",
  },
  imageFilename: {
    pattern: "my-sketch-<count>",
    digitsInCounter: 10,
    counterStartValue: 100
  }
})
```

- Pros
  - Easy to customize
- Cons
  - Low flexibility
  - Need a library such as [date-fns](https://date-fns.org/) for formatting timestamps

## Functions that return a filename

For example, something like this:

```js
P5Capture.setDefaultOptions({
  baseFilename(date) {
    return `my-sketch-${date.valueOf()}`
  },
  imageFilename(count) {
    const newCount = count + 100
    const countStr = newCount.toString().padStart(10, "0")
    return `my-sketch-${countStr}`
  }
})
```

- Pros
  - High flexibility
  - No library required
- Cons
  - Requires JavaScript skills to create complex file patterns

**Comment by tapioca24** [comment]
I prefer the function option because it is simpler and more flexible. It would meet almost all requirements.

**Comment by msoriaro** [comment]
Hi there! Thanks so much for dedicating time to this! In my opinion, the function option is also the most flexible. The user can have their own functions written with any arbitrary complexity, and for those who are not experts they can always just copy-paste an easy example for the most basic features (basically, those functions you have there in the example). Then an expert user can build upon that as much as they please. If this does not complicate your development of the library, I believe it's a useful feature for many use cases. Thank you so much once again!

**Comment by tapioca24** [comment]
@msoriaro 
Thank you for your input. It has helped me to confirm that the function option is optimal.
I am working on this now! Basically this will be ready soon. I will let you know when it''s done.

**Comment by tapioca24** [comment]
Hi, @msoriaro.

Added by https://github.com/tapioca24/p5.capture/pull/13. Released in 1.3.0. Now you can customize filenames using the `baseFilename` and `imageFilename` options!
Thank you again for your valuable suggestions. Let me know if you have any problems.
